jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/EagerContentHandler.java

```diff
@@ -0,0 +1,618 @@
+         */
+        String getApplicableMimeType();
+
+        /**
+         * @param contentType The content type of the request
+         * @param mimeType The mime type extracted from the request
+         * @param handler The next handler to call
+         * @param request The request
+         * @param response The response
+         * @param callback The callback
+         * @return An {@link ContentLoader} instance if the content can be loaded eagerly, else {@code null}.
+         */
+        ContentLoader newContentLoader(String contentType, String mimeType, Handler handler, Request request, Response response, Callback callback);
+    }
+
+    /**
+     * An Eager Content processor, created by a {@link ContentLoaderFactory} to asynchronous load content from a {@link Request}
+     * before calling the {@link Handler#handle(Request, Response, Callback)} method of the passed {@link Handler}.
+     */
+    public abstract static class ContentLoader
+    {
+        private final Handler _handler;
+        private final Request _request;
+        private final Response _response;
+        private final Callback _callback;
+
+        protected ContentLoader(Handler handler, Request request, Response response, Callback callback)
+        {
+            _handler = Objects.requireNonNull(handler);
+            _request = Objects.requireNonNull(request);
+            _response = Objects.requireNonNull(response);
+            _callback = Objects.requireNonNull(callback);
+        }
+
+        protected Handler getHandler()
+        {
+            return _handler;
+        }
+
+        protected Request getRequest()
+        {
+            return _request;
+        }
+
+        protected Response getResponse()
+        {
+            return _response;
+        }
+
+        protected Callback getCallback()
+        {
+            return _callback;
+        }
+
+        protected void handle()
+        {
+            handle(getRequest(), getResponse(), getCallback());
+        }
+
+        protected boolean handle(Request request, Response response, Callback callback)
+        {
+            try
+            {
+                if (getHandler().handle(request, response, callback))
+                    return true;
+
+                // The handle was rejected, so write the error using the original potentially unwrapped request/response/callback
+                Response.writeError(getRequest(), getResponse(), getCallback(), HttpStatus.NOT_FOUND_404);
+            }
+            catch (Throwable t)
+            {
+                // The handle failed, so write the error using the original potentially unwrapped request/response/callback
+                Response.writeError(getRequest(), getResponse(), getCallback(), t);
+            }
+            // return false to indicate the passed request/response/callback were not used.
+            return false;
```

Comment:
Cannot return `false` here, because in all cases the response is generated and the callback completed.

This return value is only used in `RetainedContentLoader.doHandle()` to release the chunks, but since the callback is invoked in all cases, they will be released already.

Perhaps we do not need a `boolean` return type, just `void`.